### PR TITLE
Monomial ordering applications

### PR DESCRIPTION
**Pull request by ederc**
1. Updates `leading_ideal` to enforce a given monomial ordering.
2. Reenables `groebner-test.jl` with new monomial ordering interface

### CODE REVIEW

**Comment by joschmitt** [comment]
It fails in "my" code because `exponent(::MPolyElem_dec, ...)` is missing. I added a commit fixing this to my fork (https://github.com/joschmitt/Oscar.jl/commit/ef72813598e9593decdaa2fb7e37b08b0c076834) so you can cherry-pick it here. (I'm not sure whether I can push to this branch and wouldn't dare.)

File: src/Rings/groebner.jl [hunk 1]
```diff
@@ -205,58 +205,57 @@ end
...
 Given an array of fitting type this function returns the ideal
-generated by the leading monomials of the given array elements.
-If not otherwise given as a further argument this is done w.r.t.
-the degree reverse lexicographical monomial ordering.
+generated by the leading monomials of the given array elements
+w.r.t. the given monomial ordering.
 """
-function leading_ideal(g::Vector{Any}, args...)
-  return leading_ideal(typeof(g[1])[ f for f in g ], args...)
+function leading_ideal(g::Vector{Any}; ordering::MonomialOrdering)
```
Comment (fingolfin) [inline src/Rings/groebner.jl]: Note that keyword arguments do *not* participate in method dispatch.

So in effect, the above method turns `ordering` into a required keyword argument. Is that the intention?

I am also confused as to why a `Vector{Any}` is expected here. I would have expected a Vector of polynomials. What "array elements" are to be used here then?

Comment (fingolfin) [reply src/Rings/groebner.jl]: Argh, of course the PR description says "enforce a given monomial ordering" so yes, you do this intentionally.

So why is it a keyword argument?

Comment (ederc) [reply src/Rings/groebner.jl]: We (@wdecker) want to have the orderings as keywords for the whole gb related stuff. It is also a keyword for `groebner_basis`and should be one for `normal_form`, etc.

Comment (ederc) [reply src/Rings/groebner.jl]: And I removed the `Vector{Any}` instantiation of `leading_ideal`, it was deprecated anyway.

**Review by thofma — APPROVED** [review]

**Pull request merged**